Prox/Prox/PlaceDetails/PlaceDetailViewController.swift

```diff
@@ -97,52 +110,242 @@ class PlaceDetailViewController: UIViewController {
+                self.currentCardViewController.cardView.addGestureRecognizer(self.panGestureRecognizer)
+                self.previousCardViewController = newPreviousCardViewController
+            }
+        })
     }
+
+
+    fileprivate func insertNewPreviousViewController(toPreviousCard previousCardViewController: PlaceDetailsCardViewController) -> PlaceDetailsCardViewController? {
+        // deactivate and remove previous view controller trailing constraint
+        if let previousCardViewTrailingConstraint = previousCardViewTrailingConstraint {
+            NSLayoutConstraint.deactivate([previousCardViewTrailingConstraint])
+        }
+
+        guard let newNext = dataSource?.previousPlace(forPlace: previousCardViewController.place) else {
+            previousCardViewTrailingConstraint = nil
+            return nil
+        }
+
+        let newPreviousCardViewController = dequeuePlaceCardViewController(forPlace:newNext)
+        self.view.addSubview(newPreviousCardViewController.cardView)
+        self.addChildViewController(newPreviousCardViewController)
+        self.previousCardViewTrailingConstraint = newPreviousCardViewController.cardView.trailingAnchor.constraint(equalTo: previousCardViewController.cardView.leadingAnchor, constant: -cardViewSpacingConstant)
+        NSLayoutConstraint.activate([newPreviousCardViewController.cardView.topAnchor.constraint(equalTo: view.topAnchor, constant: cardViewTopAnchorConstant),
+                                     newPreviousCardViewController.cardView.widthAnchor.constraint(equalToConstant: cardViewWidthConstant),
+                                     self.previousCardViewTrailingConstraint!], translatesAutoresizingMaskIntoConstraints: false)
+
+        return newPreviousCardViewController
+    }
+
+    fileprivate func setupConstraints(forNewNextCard newNextCard: PlaceDetailsCardViewController, toNewCurrentCard newCurrentCard: PlaceDetailsCardViewController) {
+        // setup constraints for new previous card
+        if let nextCardViewLeadingConstraint = nextCardViewLeadingConstraint {
+            NSLayoutConstraint.deactivate([nextCardViewLeadingConstraint])
+        }
+        self.nextCardViewLeadingConstraint = newNextCard.cardView.leadingAnchor.constraint(equalTo: newCurrentCard.cardView.trailingAnchor, constant: cardViewSpacingConstant)
+        NSLayoutConstraint.activate([self.nextCardViewLeadingConstraint!])
+    }
+
+    fileprivate func unwindToCurrentPlaceCard(animateWithDuration animationDuration: TimeInterval) {
+        currentCardViewCenterXConstraint?.constant = 0
+        UIView.animate(withDuration: animationDuration, delay: 0.0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0.0, options: .curveEaseOut, animations: {
+            self.view.layoutIfNeeded()
+            }, completion: nil )
+    }
+
+    fileprivate func addImageCarousel(forNextCard nextCard: PlaceDetailsCardViewController) -> UIView {
```

Comment:
nit: `addImageCarousel(forNewCurrentCard:...` – it's confusing with the next/previous lingo.